### ISSUE TITLE
discrepancy between `declare -A` and `local -A`

### ISSUE THREAD

**Issue by mmlb** [koalaman/shellcheck#953]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2154
- My shellcheck version (`shellcheck --version` or "online"): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/usr/bin/bash

function arrDeclare() {
  declare -A arrd
  arrd[foo]=bar
  echo "${arrd[*]}"
}

function arrLocal() {
  local -A arrl
  arrl[fool]=bar
  echo "${arrl[*]}"
}

```

#### Here's what shellcheck currently says:

```
Line 11:
  arrl[fool]=bar
       ^-- SC2154: fool is referenced but not assigned.
```

#### Here's what I wanted or expected to see:
nothing

#### More info
This is similar to #359, the difference is in how the array was declared. According to [bash docs](https://www.gnu.org/software/bash/manual/html_node/Bash-Builtins.html#index-local) `local` is synonymous with `declare` + local to function.

**Comment by Nightfirecat** [comment]
I'm able to reproduce the same in my script; changing `local` to `declare` stops SC2154 from triggering.

**Comment by PX03AFK** [comment]
Also reported in #958

**Comment by mmlb** [comment]
yay :tada:
